File: lab2/F3Thread.java
package lab2;

public class F3Thread implements Runnable {

  public static int count = 0; // количество созданных потоков
  String name; // имя потока
  Matrix P = new Matrix(1000);
  Matrix R = new Matrix(1000);
  Matrix MS = new Matrix(1000, 1000);
  Matrix MT = new Matrix(1000, 1000);
  Matrix O = new Matrix(1000, 1000);

  public F3Thread(String name, Matrix P, Matrix R, Matrix MS, Matrix MT) {
    this.name = name.toUpperCase();
    this.R = R;
    this.P = P;
    this.MS = MS;
    this.MT = MT;
  }

  public F3Thread(String name) {
    this.name = name.toUpperCase();
    P.fillRandomValues();
    R.fillRandomValues();
    MS.fillRandomValues();
    MT.fillRandomValues();
    O.fillRandomValues();
  }

  // O = (P+R)*(MS*MT)
  @Override
  public void run() {
    long start = 0, finish = 0;
    Thread.currentThread().setName(this.name);
    System.out.println();
    System.out.println("  Thread " + Thread.currentThread().getId() + " named '" + Thread.currentThread().getName() + "' is running...");
    System.out.println();
    try {
      // Displaying the thread that is running
      start = System.nanoTime();
      O = Matrix.multiply(
        Matrix.multiply(MS, MT),
        Matrix.add(P, R)
      );
      // MF.displayMatrix();
      finish = System.nanoTime();
    } catch (Exception e) {
      // Throwing an exception
      System.out.println("  Exception is caught");
    }
    System.out.println();
    System.out.println(
      "  Thread " + Thread.currentThread().getName() + " is over with time: " + Logger.getTime(finish - start)
    );
    System.out.println();
  }
}

File: lab.js-threads/f3Thread.js
'use strict';

const Matrix = require('./Matrix.js');
const helpers = require('./helpers.js');
const threads = require('worker_threads');

let { P, R, MA, MD } = threads.workerData;
let start = 0, finish = 0;
MA = new Matrix(MA.mainMatrix);
MD = new Matrix(MD.mainMatrix);
P = new Matrix(P.mainMatrix);
R = new Matrix(R.mainMatrix);
// Thread.currentThread().setName(this.name);
console.log("  Thread " + threads.threadId + " named '" + threads.name + "' is running...");
console.log();
try {
  // Displaying the thread that is running
  start = Date.now();
  const res3 = Matrix.multiply(
    Matrix.multiply(MA, MD),
    Matrix.add(P, R)
  );
  // ME.displayMatrix();
  finish = Date.now();
} catch (e) {
  // Throwing an exception
  console.log("  Exception is caught");
}
console.log();
console.log(
  "  Thread " + threads.threadId + " is over with time: " + helpers.getTime(finish, start)
);
console.log();
// process.exit(0);
